contracts/staking/ClusterSelector.sol

```diff
@@ -247,22 +247,39 @@ contract ClusterSelector is
 
     //-------------------------------- Admin functions ends --------------------------------//
 
-    function getClusters(uint256 epochNumber) public view returns (address[] memory) {
-        uint256 _nextEpoch = getCurrentEpoch() + 1;
-        // To ensure invalid data is not provided for epochs where clusters are not selected
-        require(epochNumber <= _nextEpoch, Errors.CLUSTER_SELECTION_NOT_COMPLETE);
-        if (epochNumber <= 1) {
-            return new address[](0);
-        }
-        address[] memory clusters = clustersSelected[epochNumber];
+    function _getClusters(uint256 _epoch, uint256 _nextEpoch) internal view returns (address[] memory) {
+        address[] memory clusters = clustersSelected[_epoch];
 
         if (clusters.length == 0) {
-            require(epochNumber != _nextEpoch, Errors.CLUSTER_SELECTION_NOT_COMPLETE);
-            return getClusters(epochNumber - 1);
+            require(_epoch != _nextEpoch, Errors.CLUSTER_SELECTION_NOT_COMPLETE);
```

Comment:
This check can be moved to first time it is called, as that is the only time it can be equal.

Reply:
this would never be executed in the first place under normal circumstances, doesn't seem worth it